extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/QuarkusAsyncHealthCheckFactory.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.smallrye.health.runtime;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+
+import io.quarkus.arc.AlternativePriority;
+import io.smallrye.health.AsyncHealthCheckFactory;
+import io.smallrye.health.api.AsyncHealthCheck;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.vertx.MutinyHelper;
+import io.vertx.core.Vertx;
+
+/**
+ * Quarkus specific health check factory that runs blocking and reactive
+ * health checks with different executors provided by {@link MutinyHelper}.
+ */
+@ApplicationScoped
+@AlternativePriority(1)
+public class QuarkusAsyncHealthCheckFactory extends AsyncHealthCheckFactory {
+
+    @Inject
+    Vertx vertx;
+
+    @Override
+    public Uni<HealthCheckResponse> callSync(HealthCheck healthCheck) {
+        return super.callSync(healthCheck)
+                .runSubscriptionOn(MutinyHelper.blockingExecutor(vertx));
```

Comment:
Will this actually dispatch to a different worker thread if the current thread is a worker thread? Is so, that's not ideal

Reply:
TBH, I don't know. When blocking route is used can we be sure that it will always run on worker thread? If so, then it should be enough to just remove the callSync() override.

Reply:
It is easy to test actually if blocking is allowed on the current thread, just call `io.quarkus.runtimeBlockingOperationControl.isBlockingAllowed()`.

Reply:
@geoand added the check. Thanks for the pointer.